src/devices/tuya.ts

```diff
@@ -19542,4 +19542,148 @@ export const definitions: DefinitionWithExtend[] = [
+
+            e.binary("led_switch4", ea.ALL, "ON", "OFF").withDescription("Switch of LED 4"),
+            e.numeric("led_warm4", ea.STATE_SET).withDescription("Color temperature of LED 4").withValueMin(1).withValueMax(100),
+            e.numeric("led_bright4", ea.STATE_SET).withDescription("Brightness of LED 4").withValueMin(1).withValueMax(100),
+
+            e.enum("cover_state1", ea.STATE_SET, ["open", "stop", "close"]).withDescription("State of Cover 1"),
+            e.numeric("cover_position1", ea.STATE_SET).withDescription("Position of Cover 1").withUnit("%").withValueMin(0).withValueMax(100),
+
+            e.enum("cover_state2", ea.STATE_SET, ["open", "stop", "close"]).withDescription("State of Cover 2"),
+            e.numeric("cover_position2", ea.STATE_SET).withDescription("Position of Cover 2").withUnit("%").withValueMin(0).withValueMax(100),
+
+            e.enum("cover_state3", ea.STATE_SET, ["open", "stop", "close"]).withDescription("State of Cover 3"),
+            e.numeric("cover_position3", ea.STATE_SET).withDescription("Position of Cover 3").withUnit("%").withValueMin(0).withValueMax(100),
+
+            e.enum("cover_state4", ea.STATE_SET, ["open", "stop", "close"]).withDescription("State of Cover 4"),
+            e.numeric("cover_position4", ea.STATE_SET).withDescription("Position of Cover 4").withUnit("%").withValueMin(0).withValueMax(100),
+        ],
+
+        meta: {
+            multiEndpoint: true,
+            tuyaDatapoints: [
+                [1, "action", tuya.valueConverter.static("scene_1")],
+                [2, "action", tuya.valueConverter.static("scene_2")],
+                [3, "action", tuya.valueConverter.static("scene_3")],
+                [4, "action", tuya.valueConverter.static("scene_4")],
+                [5, "action", tuya.valueConverter.static("scene_5")],
+                [6, "action", tuya.valueConverter.static("scene_6")],
+                [7, "action", tuya.valueConverter.static("scene_7")],
+                [8, "action", tuya.valueConverter.static("scene_8")],
+
+                [102, "led_bright1", tuya.valueConverter.raw],
+                [103, "led_bright2", tuya.valueConverter.raw],
+                [105, "led_bright3", tuya.valueConverter.raw],
+                [107, "led_bright4", tuya.valueConverter.raw],
+
+                [109, "led_warm1", tuya.valueConverter.raw],
+                [110, "led_warm2", tuya.valueConverter.raw],
+                [111, "led_warm3", tuya.valueConverter.raw],
+                [112, "led_warm4", tuya.valueConverter.raw],
+
+                [113, "cover_position1", tuya.valueConverter.raw],
+                [114, "cover_position2", tuya.valueConverter.raw],
+                [115, "cover_position3", tuya.valueConverter.raw],
+                [116, "cover_position4", tuya.valueConverter.raw],
+
+                [117, "led_switch1", tuya.valueConverter.onOff],
```

Comment:
```suggestion
                [117, "led_switch_l1", tuya.valueConverter.onOff],
```

Can you also add the ep names here? (same as e.g. state_l1, also do it for led_warm2 etc.

Reply:
Is it functional or a name convention to make things consistent? Thanks!

Reply:
Done. Endpoints are now used. So every post-fix is auto-generated.